chimney-macro-commons/src/main/scala-2/io/scalaland/chimney/internal/compiletime/datatypes/SealedHierarchiesPlatform.scala

```diff
@@ -36,7 +36,7 @@ trait SealedHierarchiesPlatform extends SealedHierarchies { this: DefinitionsPla
       // calling .distinct here as `knownDirectSubclasses` returns duplicates for multiply-inherited types
       extractRecursively(Type[A].tpe.typeSymbol.asType).distinct.map(typeSymbol =>
         subtypeName(typeSymbol) -> subtypeTypeOf[A](typeSymbol)
-      )
+      ).sortBy(_._1)
```

Comment:
While it would makes compilation deterministic most of the time, it might be at odds with how Scala 3 generates `.children`. So if we are improving that, I'd prefer to have **both** of the implementations use something like [file-position-based ordering](https://github.com/scalalandio/enumz/blob/master/modules/enumz/src/main/scala/io/scalaland/enumz/internal/SumTypeEnumMacros.scala#L13).

I said "most of the time", because we had cases in the past when someone compiled:

```scala
sealed trait Foo
object Foo {
  object Bar {
    case object Impl extends Foo
  }
  object Baz {
    case object Impl extends Foo
  }
}
```

which is a valid Scala, where only 2 `knownDirectSubclasses` have the same name. So ordering for such case would still be non-deterministic. Ordering by position in file has no such issues and matches what most people would expect.

Reply:
that's a great point, thank you!

Reply:
Thank you! 🙏 Could you add it here as well: https://github.com/scalalandio/chimney/blob/master/chimney-macro-commons/src/main/scala-3/io/scalaland/chimney/internal/compiletime/datatypes/SealedHierarchiesPlatform.scala#L33 ?

Just to make sure both implementation follow the same contract.

Since `symbol.pos` is `Option[Position]` I think that we could assume `None < Some(_)` and then `startLine` comparison -> `startColumn` comparison to make both the matching behavior of both implementations obvious.

Reply:
Oh, completely forgot about scala3, fixed that